src/ray/util/size_literals.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These user-defined literals makes sizes.
+
+#pragma once
+
+#include <cstdint>
+
+constexpr unsigned long long operator""_PiB(unsigned long long sz) {
+  return sz * 1024ULL * 1024ULL * 1024ULL * 1024ULL * 1024ULL;
```

Comment:
do we consider nesting implementations? this can make it more reviewable.

```cc

constexpr size_t operator""_PiB(unsigned long long sz) {
  return sz * 1024_TiB;
}
```

Reply:
Sounds good!

Reply:
Updated.